### PR TITLE
Add fixer for require-has-block-helper rule

### PR DESCRIPTION
**Pull request by patocallaghan**
Follow up to #1731 

Adds fixer for require-has-block-helper rule

Some examples of the fixer. See tests for more.

|Before|After|
|-|-|
|`{{hasBlock}}`|`{{has-block}}`|
|`{{hasBlockParams}}`|`{{has-block-params}}`|
|`{{if hasBlockParams "true" "false"}}`|`{{if (has-block-params) "true" "false"}}`|
|`{{if (hasBlockParams "inverse") "true" "false"}}`|`{{if (has-block-params "inverse") "true" "false"}}`|
|`{{#if hasBlock}}{{/if}}`|`{{#if (has-block))}}{{/if}}`|

/cc @rwjblue @pzuraq

### CODE REVIEW

File: lib/rules/require-has-block-helper.js [hunk 1]
```diff
@@ -14,14 +16,31 @@ function getErrorMessage(name) {
...
-            line: node.loc && node.loc.start.line,
-            column: node.loc && node.loc.start.column,
-            source: this.sourceForNode(node),
-          });
+      PathExpression(node, path) {
+        if (this.mode === 'fix') {
+          if (TRANSFORMATIONS[node.original]) {
+            let parent = path.parent;
+            if (
+              parent &&
```
Comment (rwjblue) [inline lib/rules/require-has-block-helper.js:24]: When is `parent` not truthy? I think it always has to be... 🤔

File: lib/rules/require-has-block-helper.js [hunk 2]
```diff
@@ -14,14 +16,31 @@ function getErrorMessage(name) {
...
+              (parent.node.type === 'BlockStatement' ||
+                (['MustacheStatement', 'SubExpression'].includes(parent.node.type) &&
+                  parent.node.path.original !== node.original))
+            ) {
```
Comment (rwjblue) [inline lib/rules/require-has-block-helper.js:28]: Let's break this conditional down into a few variables, to help simplify the mental model.

```suggestion
           let isBlockStatement = parent.node.type === 'BlockStatement';
           let isImplicitInvocation = ['MustacheStatement', 'SubExpression'].includes(parent.node.type) && parent.node.path.original !== node.original);
                  
            if (isBlockStatement || isImplicitInvocation) {
```

**Review by rwjblue — COMMENTED** [review]
Looks good, one minor suggestion / tweak and should be good to go...

File: lib/rules/require-has-block-helper.js (hunk 1 above)
Comment (patocallaghan) [reply lib/rules/require-has-block-helper.js]: Yep, you're right 👍

File: lib/rules/require-has-block-helper.js (hunk 2 above)
Comment (patocallaghan) [reply lib/rules/require-has-block-helper.js]: Thanks, reads much better 👍

**Review by rwjblue — APPROVED** [review]
Thank you!!

**Pull request merged**